modules/tinymce/src/plugins/table/main/ts/ui/Buttons.ts

```diff
@@ -207,6 +208,27 @@ const addButtons = (editor: Editor, selectionTargets: SelectionTargets, clipboar
       onSetup: selectionTargets.onSetupCellOrRow
     });
   }
+
+  editor.ui.registry.addMenuButton('tablecellvalign', {
+    icon: 'vertical-align',
+    tooltip: 'Vertical align',
+    fetch: (callback) => {
```

Comment:
This largely duplicates the `makeClassFetch` with the exception of the value being passed in the command and the `text`/`title` name difference. Please refactor this to make that more generic and to try to remove some duplication (I imagine you're going to need more similar things moving forward).

An example would be doing something like:
```
const makeFetch = <T>(editor: Editor, items: T[], format: string, extractText: (item: T) => string, extractValue: (item: T) => string | Record<string, string>) => {
...
};
```
or:
```
const makeFetch = <T>(editor: Editor, items: T[], command: string, format: string, extractText: (item: T) => string, onAction: (item: T) => void) => {
...
};
```

Another improvement would be to change `verticalAlignValues` to use `title` instead of `text` so it has a common interface and that way you wouldn't need to have the `extractText` function passed in.